doc/sphinx/language/gallina-extensions.rst

```diff
@@ -20,13 +20,6 @@ expressions. In this sense, the :cmd:`Record` construction allows defining
 
 .. _record_grammar:
 
-  .. productionlist:: sentence
-     record         : `record_keyword` `record_body` with … with `record_body`
-     record_keyword : Record | Inductive | CoInductive
-     record_body    : `ident` [ `binders` ] [: `sort` ] := [ `ident` ] { [ `field` ; … ; `field` ] }.
-     field          : `ident` [ `binders` ] : `type` [ `decl_notations` ]
-                    : `ident` [ `binders` ] [: `type` ] := `term`
-
 .. cmd:: {| Record | Structure } @inductive_definition {* with @inductive_definition }
    :name: Record; Structure
```

Comment:
Since the recent changes, the first sentence of description doesn't make sense anymore. It would be good to fix it at the lowest possible cost (since this documentation is anyway going to be refactored soon).

Reply:
If the `type` must be a `sort`, should we update the mlg to reflect that?  Maybe applies in other points in the doc?

Reply:
Also, are both of these forms valid for a `Record`?  If not, should mention that.

![image](https://user-images.githubusercontent.com/1253341/76354430-ab4e2900-62cf-11ea-8f57-a77c1c3aa9e0.png)

Reply:
Check what I wrote--probably not entirely correct/complete yet.

Reply:
Can you have universes on the record `ident`?

Reply:
- It doesn't have to be a `sort`. The grammar that you removed was artificially restrictive.

- For records, only the second construct is valid. The other one gives `The Record keyword is for types defined using the syntax { ... }.` Yes, it should be written down, and even the error message itself could be documented.

- If you mean the `ident` from the `ident { record_fields }` construct, then this is just the name of the record builder and its universes are likely to be exactly the same as the type itself, so there is probably no need to repeat the possibility of providing those universes.

Reply:
It seems to me that `inductive_definition` has been reused for several similar but distinct commands making the syntax a bit over-general.  Perhaps we should consider creating separate nonterminals in the doc, e.g. one for records and one for the case using constructors.  
Then the nonterminal for Record could omit the {? %| } {+| @constructor } entirely.

That would also simplify the first paragraph "Each inductive_definition defines a record...".

Reply:
You can create inductive and co-inductive records with the `Inductive` or `CoInductive` keyword and the record syntax. That's why the two have been mixed. However, it should indeed be possible to separate the two syntax a bit more properly.

Reply:
Cf. #11868. After digging into this, I am convinced that, at least at the documentation level, if not at the parsing level, we must do this split to better explain what is supported and what is not.

In particular, we should start by explaining `Variant` and `Record`, which both support a very limited syntax (they do support mutual definitions, but they make no sense, `Variant` only accept the constructors syntax and `Record` the record syntax).

Note that I'm talking about the refactored refman here, and all of this is out of scope for the present PR.

Reply:
In the meantime, let's revert this description to `{| Record | Structure } {? > } @ident_decl {* @binder } {? : @type } {? := {? ident } { {*; record_field } } {? decl_notations } }` and let's move the definition of `record_field` and `field_body` here.

Reply:
Please check what I did for Record and Variant, close but slightly different from your recommendation.